File: src/models/investigator.ts
import investigatorsData from '../data/investigators.json'

export interface Investigator {
    id: string;
    name: string;
    subName: string;
    faction: string;
    set: string;
}

export function loadInvestigatorMap(): Map<string, Investigator[]> {
    const result: Map<string, Investigator[]> = new Map<string, Investigator[]>();

    (investigatorsData as Investigator[]).forEach((investigator) => {
      if (result.has(investigator.name)) {
        result.get(investigator.name)?.push(investigator);
      } else {
        result.set(investigator.name, [investigator]);
      }
    });

    return result;
}

File: client/ArkhamDB.Scraper/Models/Investigator.cs
namespace ArkhamDB.Scraper.Models
{
    using System.Text.Json.Serialization;

    internal class Investigator
    {
        public string Id { get; }

        public string Name { get; }

        public string SubName { get; }

        public Faction Faction { get; }

        public string Set { get;  }

        public Investigator(string id, string name, string subName, Faction faction, string set)
        {
            Id = id;
            Name = name;
            SubName = subName;
            Faction = faction;
            Set = set;
        }
    }
}
